### PR TITLE
feat: replace QA_ENV with TWO_FA_ENABLED and remove demo scope

### PR DESCRIPTION
**Pull request by Nil20**
> [!NOTE]
> Currently, we do **not** run e2e tests as a check on `opencrvs-countryconfig`-repo PRs. Please ensure your PR doesn't break any e2e tests.
>
> One method for doing this is to open a PR with these changes to `opencrvs-farajaland` as well, and see if the PR check passes there.

## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: src/data-seeding/employees/handler.ts [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
 import { readCSVToJSON } from '@countryconfig/utils'
 import { Request, ResponseToolkit } from '@hapi/hapi'
 
 export async function usersHandler(_: Request, h: ResponseToolkit) {
+  // TWO_FA_ENABLED is used as a proxy for "real production environment".
+  // QA/staging deployments disable 2FA for testing convenience, so when it's
+  // false we seed default (test) employees instead of production ones.
   const users: unknown[] = await readCSVToJSON(
-    process.env.NODE_ENV === 'production' && !QA_ENV
+    process.env.NODE_ENV === 'production' && TWO_FA_ENABLED
```
Comment (rikukissa) [inline src/data-seeding/employees/handler.ts]: Add a new env variable for this for instance `OPENCRVS_ENVIRONMENT` where we can then have anything like `qa`, `demo`, `dev`, `production`.. You can make the whole condition then just 

```
['staging', 'production'].contains(OPENCRVS_ENVIRONMENT)
```

Or similar

File: Dockerfile.assets [hunk 2]
```diff
@@ -9,7 +9,7 @@ COPY --chmod=755 infrastructure/metabase /assets/metabase
 # Postgres
 COPY --chmod=755 infrastructure/postgres /assets/postgres
 # Elasticsearch re-indexing
-COPY --chmod=755 infrastructure/deploy/reindex.sh /assets/elasticsearch/reindex.sh
+COPY --chmod=755 infrastructure/deployment/reindex.sh /assets/elasticsearch/reindex.sh
```
Comment (rikukissa) [inline Dockerfile.assets:12]: Intentional?

Comment (Nil20) [reply Dockerfile.assets:12]: Yeah. I think the directory name changed in some PR.

https://github.com/opencrvs/opencrvs-countryconfig/blob/develop/infrastructure/deployment/reindex.sh

Broken workflow: https://github.com/opencrvs/opencrvs-farajaland/actions/runs/23939658108/job/69823030106

Also would be good to have @adskyiproger's approval as well!

**Review by rikukissa — APPROVED** [review]
